lib/ansible/modules/cloud/podman/podman_container_info.py

```diff
@@ -0,0 +1,370 @@
+                    "GroupAdd": [],
+                    "IpcMode": "",
+                    "Cgroup": "",
+                    "Links": null,
+                    "OomScoreAdj": 0,
+                    "PidMode": "",
+                    "Privileged": false,
+                    "PublishAllPorts": false,
+                    "ReadonlyRootfs": false,
+                    "SecurityOpt": [],
+                    "Tmpfs": {},
+                    "UTSMode": "",
+                    "UsernsMode": "",
+                    "ShmSize": 65536000,
+                    "Runtime": "oci",
+                    "ConsoleSize": [
+                        0,
+                        0
+                    ],
+                    "Isolation": "",
+                    "CpuShares": 0,
+                    "Memory": 0,
+                    "NanoCpus": 0,
+                    "CgroupParent": "",
+                    "BlkioWeight": 0,
+                    "BlkioWeightDevice": null,
+                    "BlkioDeviceReadBps": null,
+                    "BlkioDeviceWriteBps": null,
+                    "BlkioDeviceReadIOps": null,
+                    "BlkioDeviceWriteIOps": null,
+                    "CpuPeriod": 0,
+                    "CpuQuota": 0,
+                    "CpuRealtimePeriod": 0,
+                    "CpuRealtimeRuntime": 0,
+                    "CpusetCpus": "",
+                    "CpusetMems": "",
+                    "Devices": [],
+                    "DiskQuota": 0,
+                    "KernelMemory": 0,
+                    "MemoryReservation": 0,
+                    "MemorySwap": 0,
+                    "MemorySwappiness": -1,
+                    "OomKillDisable": false,
+                    "PidsLimit": 0,
+                    "Ulimits": [
+                        {
+                            "Name": "RLIMIT_NOFILE",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        },
+                        {
+                            "Name": "RLIMIT_NPROC",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        }
+                    ],
+                    "CpuCount": 0,
+                    "CpuPercent": 0,
+                    "IOMaximumIOps": 0,
+                    "IOMaximumBandwidth": 0
+                }
+                }
+        ]
+"""
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_containers_facts(module, executable, name):
+    if not name:
+        all_names = [executable, 'container', 'ls', '-q', '-a']
+        rc, out, err = module.run_command(all_names)
+        if rc != 0:
+            module.fail_json(msg="Unable to get list of containers: %s" % err)
+        name = out.split()
+        if not name:
+            return [], out, err
+    command = [executable, 'container', 'inspect']
+    command.extend(name)
+    rc, out, err = module.run_command(command)
+    if rc != 0:
```

Comment:
I think though, maybe we shouldn't fail a module when no container with such name, but just skip it in the output? docker_container_info doesn't fail (and seems for me logical to not fail info module). It just returns an empty result.
We also made this for podman_image_info module: https://github.com/ansible/ansible/issues/57899
Maybe worth to adopt the same logic here and in podman_volume_info modules? To return info only about existing volumes/images/containers, and just ignore non existent. WDYT, @felixfontein @samdoran and others?

Reply:
Simply skipping non-existing containers would be good behavior IMO. There are two problems, though:
 * if you specify multiple containers and one of them does not exist, you get an error and no result;
 * if you get a "serious" error (i.e. something other than "no such container", you still want to fail.

These problems are definitely solvable, the main question is probably how :) IMO there's a problem with the approach in #57962, which is not that important for images, but a lot more important for containers, especially in environment where there are a lot of containers and where they are rather short-lived: it can happen that they went away between the existence check and the inspection.

How about not going through them one-by-one to check for existance, and then inspect them all, but how about going through them and inspecting them one-by-one?

Reply:
We can do different: to gather info for all containers by default, and then if there's set of names - to filter output for these names. If this name is not in collected info - skip it.

Reply:
How do you gather information for all containers at once?

Reply:
Well, as I do it in the module:
`podman container ls -q -a` and `podman inspect ` for `output.split()`:
`podman container inspect name1 name2 name3`

If we check for existence:
`podman container exists name1` and then `podman container inspect name1` 
then delay between commands `podman container exists name1` and `podman container inspect name1` is the same as between `podman container ls -q -a` and `podman container inspect name1 name2 name3`.
We can miss container in the same way.

So if have `names` specified, we can do something like:
`podman container ls -q -a` and then filter output like:
`print [i for i in output if i['Name'] in names]`
If container not in `names` it's ignored, if it's in `names` and doesn't exist, it's ignored as well.
The only possible danger I see here if output of inspection will change and we can't filter by name. Or it'll be lower case (but this I can work out).

Reply:
Since these have the same drawbacks, why not simply iterate through the container list and inspect every single one? Then for every single call, you can easily distinguish between "container doesn't exist (anymore)" and "here's the inspection result, so it obviously exists".

Reply:
If I have 10 containers it will be very expensive to check each one after one, instead of 1 inspect command for all of them. Actually it will be only 2 commands - `ps` and `inspect`. No matter how many containers names are passed.

Reply:
That's true. So how about:
 1) Use `podman ps ...` to get a list and filter by it;
 2) Use `podman inspect <names>` to inspect;
 3) If the previous command fails, fall back to iterate over the names list and `inspect` them one-by-one.
Then it will always work, and only needs to fall back to the slow version if necessary. (You coul also do a divide and conquer approach by splitting the list in half and recursively work on each half, but that can also be added later if someone complains the default approach is too slow.)

Reply:
I'm not sure about iterating over containers. if environment is so fast changing that second command failed, most likely then most of next commands will fail as well. Maybe we can just add retries with `podman ps`, will be more efficient. If inspection fails again, well this is a bad luck, the delay between commands is in microseconds resolution. Collecting info from such unstable environment doesn't make much sense.
I'd suggest to retry once for now if we feel it would be safer. And in next PRs we can introduce something more complex if we need. Or maybe to leave such improvements to next PRs, it's really extreme case.